design/kubebuilder-migration.md

```diff
@@ -0,0 +1,64 @@
+# Move Contour to use kubebuilder instead of informers
+
+Status: Draft
+
+Contour's method of interaction with Kubernetes is two years old, and the community has moved on from Informers.
+This proposal suggests that we should move to using Kubebuilder [controller-runtime][1] for interacting with Kubernetes.
+This will make extending Contour with additional CRDs as we need to more straightforward.
+
+## Goals
+
+- Migrate Contour to use Kubebuilder [controller-runtime][1] instead of informers.
+
+## Non Goals
+
+- Change anything after the DAG builder
```

Comment:
probably redundant, this could be moved to the high level design section with a note that we're just considering how we get notifications from the api server. period.